test/integration/graphql_test.lua

```diff
@@ -595,6 +625,15 @@ g.test_custom_type_scalar_variables = function()
             callback = 'test.test_custom_type_scalar',
         })
 
+        graphql.add_callback({
+            name = 'test_json_type',
+            args = {
+                field = json_type,
+            },
+            kind = json_type,
+            callback = 'test.test_custom_type_scalar',
```

Comment:
I think we need a bit different callback for tests. Currently it works as echo. But internally we should check that our variable in JSON representation was decoded to map-array-number-etc.

We should test passing this value via variables and inline specification (for primitive types only I think).

Reply:
This test will be failed if serialize function is not called:
https://github.com/tarantool/cartridge/blob/49349e48637e7161d48cfaf8b6bbcafbdaf92eef/test/integration/graphql_test.lua#L679-L690

This test will be failed if parseValue function is not called (if serialize was called):
https://github.com/tarantool/cartridge/blob/49349e48637e7161d48cfaf8b6bbcafbdaf92eef/test/integration/graphql_test.lua#L666-L677

This test will be failed if parseLiteral function is not called (if serialize was called):
https://github.com/tarantool/cartridge/blob/49349e48637e7161d48cfaf8b6bbcafbdaf92eef/test/integration/graphql_test.lua#L692-L703

So, I think it's not necessary to check value in callback

Reply:
But you pass a string and get a string. You don't check that happens inside.

`[input] -> [black box] -> [output]`

input is a string
output is a string

Change json.decode/encode in [black box] to "tostring" and you will get the same result, isn't it?

Reply:
Yes, if you will use tostring function, it will be passed. But if one of parsing function not used it will be failed. But I can add check to make this test readable

Reply:
Yes, it's not enough to test an interface in this case, you need to test internals.